### PR TITLE
Fixup to Origin/mike/startlette dependency

### PR DESCRIPTION
**Pull request by joshmgrant**
Trying to fix up #149. Please let me know if this not applicable or generally frowned upon.

### CODE REVIEW

**Comment by joshmgrant** [comment]
Any updates on this? I'm still happy to help :)

**Comment by robbkidd** [comment]
To appease the `lint-2.7` CI job, you can add `starlette` to [the list of directories and files for the Python 2.7 linting to ignore](https://github.com/honeycombio/beeline-python/blob/42cbb9df606dbe76535085be290f5c742c5c8743/.circleci/config.yml#L24).

Because I had trouble pushing to this PR's branch: 

<details><summary>A Patch, if you like</summary>

```patch
From 786845ae36a01b5fe0711eddeadf4b294eaea10b Mon Sep 17 00:00:00 2001
From: Robb Kidd <robbkidd@honeycomb.io>
Date: Thu, 27 May 2021 13:32:36 -0400
Subject: [PATCH] ignore starlette when linting with Py2.7

starlette supports only Python 3.6 and above. We don't have to lint this
middleware for Python 2.7, so add it to the growing 2.7 ignore list.
---
 .circleci/config.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.circleci/config.yml b/.circleci/config.yml
index 1821f71..ebf983d 100644
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -21,7 +21,7 @@ jobs:
               - "2.7"
               - << parameters.python-version >>
           steps:
-            - run: poetry run pylint --ignore=aiotrace.py,test_async.py,urllib.py,test_urllib.py --rcfile=pylint.rc beeline
+            - run: poetry run pylint --ignore=starlette,aiotrace.py,test_async.py,urllib.py,test_urllib.py --rcfile=pylint.rc beeline
       - when:
           condition:
             not:
```

</details>

File: beeline/trace.py [hunk 1]
```diff
@@ -333,7 +333,7 @@ def _run_hooks_and_send(self, span):
 
 class SynchronousTracer(Tracer):
     def __init__(self, client):
-        super(SynchronousTracer, self).__init__(client)
+        Tracer.__init__(self, client)  # noqa
```
Comment (robbkidd) [inline beeline/trace.py:336]: It'd be good to leave ourselves a reminder to Super Properly in the (hopefully near) future.

```suggestion
        Tracer.__init__(self, client)  # noqa TODO: use a Python 3.0 style super when 2.7 support is dropped
```

**Comment by robbkidd** [comment]
@MikeGoldsmith @dsoo What do y'all think of this PR supplanting #149?

File: beeline/trace.py (hunk 1 above)
Comment (joshmgrant) [reply beeline/trace.py:336]: lgtm!

**Comment by robbkidd** [comment]
@joshmgrant I've also been trying to get the various pieces of adding starlette to combine on the other PR (#149). I've pulled your changes into it, too, while tinkering.

**Comment by joshmgrant** [comment]
I am good with whatever PRs/changes get in, however it gets in @robbkidd :)

**Comment by paulosman** [comment]
Closing this. Starlette requires Python 3.6 at a minimum and the Honeycomb beeline still supports Python 2.7 and 3.5. Getting Starlette tests to pass on recent versions as well as older versions of Python is taking a lot of effort. While I understand that this would be great to see for Starlette users, for the time being, we recommend using [opentelemetry-python](https://github.com/open-telemetry/opentelemetry-python) which has [starlette instrumentation](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-starlette). 

We may resurrect this if and when we consider dropping support for older Python versions, which would be a major release.

**Comment by RZachLamberty** [comment]
@paulosman any chance this could be re-opened and re-considered now that the python version support has been updated?

**Pull request closed without merging**